src/main/scala/beam/utils/scenario/UrbanSimScenarioLoader.scala

```diff
@@ -218,6 +230,20 @@ class UrbanSimScenarioLoader(
         beamScenario.privateVehicles.put(beamVehicle.id, beamVehicle)
         vehicleCounter = vehicleCounter + 1
       }
+      if ( beamScenario.beamConfig.beam.agentsim.agents.vehicles.vehicleAdjustmentMethod.equals("STATIC_FROM_FILE")) {
+        // populate private vehicles here
+        val vehicles = readers.BeamCsvScenarioReader.readVehiclesFile(beamScenario.beamConfig.beam.agentsim.agents.vehicles.vehiclesFilePath)
+          .filter(x => Id.create(x.householdId, classOf[Household]).equals(id));
+        for {
+          vehicleInfo <- vehicles
+          vehicle = buildBeamVehicle(beamScenario.vehicleTypes, vehicleInfo, rand.nextInt)
+        } {
+          beamScenario.privateVehicles.put(vehicle.id, vehicle)
+          vehicleIds.add(vehicle.id)
+          vehicleCounter = vehicleCounter + 1
+        }
+      }
+
```

Comment:
My suggestion to make things simpler is following:
1. create an implementation of VehiclesAdjustment - FromFileVehiclesAdjustment and read vehicle types once during initialization of the vehicle adjustment object (reading a file for each household is not a good thing - the file might be big and households plenty)
2. add an optional parameter into VehiclesAdjustment.sampleVehicleTypesForHousehold - householdId
3. for FromFileVehicleAdjustment use the householdId to return appropriate vehicle types, for the rest of vehicle adjustment classes - ignore householdId